sycl/doc/extensions/proposed/sycl_ext_intel_cslice.asciidoc

```diff
@@ -0,0 +1,273 @@
+is not currently enabled by default, so it is considered an advanced feature
+which most applications are not expected to use.  This extension provides a way
+for these advanced applications to partition a device by cslice when it is
+enabled in the device driver.
+
+Unlike "tile" partitions, a cslice partition does not have any different cache
+affinity from its sibling cslice partitions.  Therefore, this extension does
+not expose this type of partitioning through
+`info::partition_property::partition_by_affinity_domain`.  Instead, it adds a
+new partitioning type
+`info::partition_property::ext_intel_partition_by_cslice`.
+
+The only Intel GPU device that currently supports this type of partitioning is
+PVC, and this support is only available when the device driver is configured in
+{multi-CCS}[multi-CCS] mode.  See that documentation for instructions on how to
+enable this mode and for other important information.  Currently, it is only
+possible to partition a device by cslice if the driver is in "2 CCS Mode" or
+"4 CCS Mode".  When in 2 CCS Mode, a tile can be partitioned into two cslice
+sub-devices.  When in 4 CCS Mode, a tile can be partitioned into four cslice
+sub-devices.
+
+This type of partitioning is currently supported only at the "tile" level.
+Therefore, a device must first be partitioned into per-tile sub-devices via
+`partition_by_affinity_domain`, and then each of the resulting sub-devices can
+be further partitioned by `ext_intel_partition_by_cslice`.
+
+It is important to understand that the device driver virtualizes work
+submission to the cslice sub-devices.  This virtualization happens only between
+processes, and not within a single process.  For example, consider a single
+process that constructs two SYCL queues on cslice sub-device #0.  Kernels
+submitted to these two queues are guaranteed to conflict, both using the same
+set of execution units.  Therefore, if a single process wants to explicitly
+submit kernels to cslice sub-devices and it wants to avoid conflict, it should
+create queues on different sub-devices.  By contrast, consider an example where
+two separate processes create a SYCL queue on cslice sub-device #0.  In this
+case, the device driver virtualizes access to this cslice, and kernels
+submitted from the first process may run on different execution units than
+kernels submitted from the second process.  In this second case, the device
+driver binds the process's requested cslice to a physical cslice according to
+the overall system load.
+
+Note that this extension can be supported by any backend and any device.  If a
+backend or device does not support the concept of cslice partitions, it can
+still conform to this extension by declaring the new enumerator and member
+functions specified below.  If the info descriptor query
+`info::device::partition_properties` does not report
+`ext_intel_partition_by_cslice`, then the implementation need not support
+the creation of cslice partitions.
```

Comment:
Is this the behavior that we want?  It seems odd to me that a backend or device would advertise support for the extension but then would NOT support the new partition property.

Reply:
An extension is either supported or not supported by the entire SYCL implementation, which may include several backends.  It would certainly make sense for an implementation to advertise support for this extension and only allow partitioning into cslices on some devices / backends.

Personally, I think it could make sense for an implementation to support this extension even if none of its backends supported this type of partitioning.  For example, this would allow the implementation to consume applications that dynamically check to see if a device supports cslice partitions.  However, that's not something we need to think deeply about since DPC++ would support this type of partitioning.  If other vendors choose to support this extension, they can decided for themselves.

Reply:
Thanks, your description makes sense to me but I still think the way this paragraph uses the terms "device" and "implementation" makes it easy to mis-understand the intent.  How about something more like this instead?

```suggestion
Note that this extension can be supported by any implementation.  If an implementation supports a
backend or device without the concept of cslice partitions it can
still conform to this extension by declaring the new enumerator and member
functions specified below.  If the info descriptor query
`info::device::partition_properties` does not report
`ext_intel_partition_by_cslice`, then the backend or device does not support
the creation of cslice partitions.
```

Reply:
Thanks, I like that.  Incorporated in 0eb19c64d9ee62af7b497ef2aa81a52442b0370d and 81b6ea1c8fe23b79e3a56ceb8cf09a1ab8101fa3